src/k8s/pkg/utils/vals/pointer.go

```diff
@@ -0,0 +1,13 @@
+package vals
+
+func Pointer[T any](v T) *T {
+	return &v
+}
+
+func Deref[T any](p *T) T {
+	if p == nil {
+		var zero T
+		return zero
+	}
+	return *p
+}
```

Comment:
This is only used for optional `bool` values, make this clear:

```suggestion
func OptionalBool(v *bool, defaultValue bool) T {
	if v != nil {
	   return *v
	}
	return defaultValue
}
```